eth_utils/address.py

```diff
@@ -73,9 +73,9 @@ def to_normalized_address(value: AnyStr) -> HexAddress:
     """
     Converts an address to its normalized hexadecimal representation.
     """
-    hex_address = hexstr_if_str(to_hex, value).lower()
+    hex_address = hexstr_if_str(to_hex, value).lower()  # type: ignore
```

Comment:
`hexstr_if_str` should be allowed to return an int, correct? Since `.lower()` throws on an int, should we put a check here? Though I guess `value` will always be one of `AnyStr`s

Reply:
Yes, `hexstr_if_str` *can* return what we refer to as the `primative` type which depending on context can be `int` and/or `bytes` and/or `bool`.

Rather than a type check, maybe a `try/except` which catches the `AttributeError` and raises a `TypeError` indicating `value` was of the wrong type?

You are correct that `value` should *always* be `AnyStr` but that doesn't mean that users won't pass in non-string values.